packages/inventory/src/ducks/inventory.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  fetchActionCreatorFactory,
+  removeActionCreatorFactory,
+  setTotalRecordsFactory,
+  reducerFactory,
+  getItemsByIdFactory,
+  getItemsArrayFactory,
+  getItemByIdFactory,
+  getTotalRecordsFactory,
+} from '@opensrp/reducer-factory';
+
+/** interface inventory */
+export interface Inventory {
+  deliveryDate: string;
+  donor: string;
+  poNumber: number;
+  productName: string;
+  providerId: string;
+  quantity: number;
+  serialNumber: string;
+  servicePointId: string;
+  stockId: string;
+  unicefSection: string;
+}
+
+/** reducer name */
+export const reducerName = 'inventory';
+
+/** Action types */
+const customfetchedActionType = 'inventory/INVENTORIES_FETCHED';
+const customRemoveActionType = 'inventory/REMOVE_INVENTORIES';
+const customSetTotalRecordsActionType = 'inventory/SET_TOTAL_INVENTORIES';
+
+/** Item Reducer */
+const reducer = reducerFactory<Inventory>(
+  reducerName,
+  customfetchedActionType,
+  customRemoveActionType,
+  customSetTotalRecordsActionType
+);
+
+// action
+/** actionCreator returns action to to add Item records to store */
+export const fetchInventories = fetchActionCreatorFactory<Inventory>(reducerName, 'servicePointId');
+export const removeInventories = removeActionCreatorFactory(reducerName);
+export const setTotalInventories = setTotalRecordsFactory(reducerName);
+
+// selectors
+export const getInventoriesById = getItemsByIdFactory<Inventory>(reducerName);
+export const getInventoryById = getItemByIdFactory<Inventory>(reducerName);
+export const getInventoriesArray = getItemsArrayFactory<Inventory>(reducerName);
+export const getTotalInventories = getTotalRecordsFactory(reducerName);
```

Comment:
we should ideally use [reselect](https://www.npmjs.com/package/reselect) for this, [e.g](https://github.com/OpenSRP/web/blob/cac8f85f8f7fc9cd32aa9a94a08e8ecb10e070ad/packages/location-management/src/ducks/locationHierarchy/index.ts#L113)